src/react-components/avatar-preview.js

```diff
@@ -148,8 +147,24 @@ export default class AvatarPreview extends Component {
   })();
 
   componentWillUnmount = () => {
+    // Gotta be particularly careful about disposing things here since we will likely create many avatar
+    // previews during a session and Chrome will eventually discard the oldest webgl context if we leak
+    // contexts by holding on to them directly or indirectly.
+
     this.scene && this.scene.traverse(disposeNode);
-    this.previewRenderer && this.previewRenderer.dispose();
+    if (this.previewRenderer) {
+      this.previewRenderer.dispose();
+      this.previewRenderer.context = null;
```

Comment:
Surprised this doesn't just get handled by GC. Is this just to push the GC along? If not, wonder if something else might still be holding onto the renderer...

Reply:
I think it does get handled by GC, I just wanted to be more certain about all the context references.
Interestingly, it seems that Chrome ended up hanging on to the contexts if you didn't dispose the OrbitControls, not only because orbit controls references the canvas, but because it attaches event listeners to the canvas. So that was the primary leak in Chrome at least.